### PR TITLE
Build with Go 1.20

### PR DESCRIPTION
**Pull request by sylr**
* [x] I added CHANGELOG entry for this change.
* [ ] Change is not relevant to the end user.

### CODE REVIEW

**Comment by sylr** [comment]
@bwplotka @GiedriusS I tried to debug the failing test locally and it turns out that the chunk size varies at each run.

As I have no understanding of prometheus storage backend I'd like to hand over this to someone competent in the matter.

**Comment by saswatamcode** [comment]
Hmm I'm not sure why `TestUpload` consistently fails here. It passes locally. Will take a deeper look

**Comment by saswatamcode** [comment]
Ok I was able to reproduce this after upgrading to Go 1.20.1 locally. The reason `TestUpload` has become nondeterministic seems to be due to the way we create blocks for testing. We use `rand.Float64()` for the samples: https://github.com/thanos-io/thanos/blob/5e3ecf67ef82c2b54e7d4bdffae2c9e5c30b6bec/pkg/testutil/e2eutil/prometheus.go#L509 which was deterministic and always produced the same random values in the same order

But as of Go 1.20 "math/rand" package automatically seeds the global random number generator which causes our block chunk file size to vary: https://tip.golang.org/doc/go1.20#minor_library_changes

I think we just need to seed rand here 🙂

**Comment by sylr** [comment]
@saswatamcode thank you!

File: pkg/testutil/e2eutil/prometheus.go [hunk 1]
```diff
@@ -484,6 +484,7 @@ func createBlock(
 	var g errgroup.Group
 	var timeStepSize = (maxt - mint) / int64(numSamples+1)
 	var batchSize = len(series) / runtime.GOMAXPROCS(0)
+	r := rand.New(rand.NewSource(int64(numSamples)))
```
[the remark this answers is not in the document]
Comment (saswatamcode) [reply pkg/testutil/e2eutil/prometheus.go:487]: This is for a test, so it's fine.

**Review by saswatamcode — APPROVED** [review]
LGTM!

**Review by GiedriusS — APPROVED** [review]
:1st_place_medal:

**Pull request merged**